clientconn_test.go

```diff
@@ -67,6 +68,26 @@ func TestTLSDialTimeout(t *testing.T) {
 	}
 }
 
+func TestDialWithContextCancel(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan struct{})
+	go func() {
```

Comment:
Calling `cancel()` asynchronously seemed like a good idea. I could call cancel immediately, but it seemed worthwhile to give `DialWithContext()` an execution pathway to success which we verify never occurs.